keydata.go

```diff
@@ -52,6 +55,92 @@ const (
 	AuthModePIN
 )
 
+type afSplitDataRawHdr struct {
+	Stripes uint32
+	HashAlg tpm2.HashAlgorithmId
+	Size    uint32
+}
+
+type afSplitDataRaw struct {
+	Hdr  afSplitDataRawHdr
+	Data tpm2.RawBytes
+}
+
+func (d *afSplitDataRaw) Marshal(w io.Writer) (nbytes int, err error) {
+	return tpm2.MarshalToWriter(w, d.Hdr, d.Data)
+}
+
+func (d *afSplitDataRaw) Unmarshal(r io.Reader) (nbytes int, err error) {
+	var h afSplitDataRawHdr
+	n, err := tpm2.UnmarshalFromReader(r, &h)
+	nbytes += n
+	if err != nil {
+		return nbytes, xerrors.Errorf("cannot unmarshal header: %w", err)
+	}
+
+	data := make([]byte, h.Size)
+	n, err = io.ReadFull(r, data)
+	nbytes += n
+	if err != nil {
+		return nbytes, xerrors.Errorf("cannot read data: %w", err)
+	}
+
+	d.Hdr = h
+	d.Data = data
+	return
+}
+
+func (d *afSplitDataRaw) data() *afSplitData {
+	return &afSplitData{
+		stripes: d.Hdr.Stripes,
+		hashAlg: d.Hdr.HashAlg,
+		data:    d.Data}
+}
+
+func makeAfSplitDataRaw(d *afSplitData) *afSplitDataRaw {
+	return &afSplitDataRaw{
+		Hdr: afSplitDataRawHdr{
+			Stripes: d.stripes,
+			HashAlg: d.hashAlg,
+			Size:    uint32(len(d.data))},
+		Data: d.data}
+}
+
+type afSplitData struct {
+	stripes uint32
+	hashAlg tpm2.HashAlgorithmId
+	data    []byte
+}
+
+func (d *afSplitData) merge() ([]byte, error) {
+	if d.stripes < 1 {
+		return nil, errors.New("invalid number of stripes")
+	}
+	if !d.hashAlg.Supported() {
+		return nil, errors.New("unsupported digest algorithm")
+	}
+	return afis.MergeHash(d.data, int(d.stripes), func() hash.Hash { return d.hashAlg.NewHash() })
+}
+
+func makeAfSplitData(data []byte, minSz int, hashAlg tpm2.HashAlgorithmId) (*afSplitData, error) {
+	stripes := uint32((minSz / len(data)) + 1)
```

Comment:
Here's the potential divide-by-zero case

Reply:
This shouldn't be possible - it's called during marshalling from keyData.Marshal, and the data length should always be non-zero:

https://github.com/snapcore/secboot/blob/fc07b3864d05363e8d253157164d89c81094f0fb/keydata.go#L296